dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -269,14 +270,21 @@ private void writeField(FieldDef fd, Object fieldValue) throws Exception
             case INTEGER:
             {
                 Long value = null;
-                if (fieldValue instanceof Long)
+                if (fieldValue==null) {
+                    value=new Long(0);
+                }
+                else if (fieldValue instanceof Long)
                 {
                     value = (Long) fieldValue;
                 }
                 else if (fieldValue instanceof Integer)
                 {
                     value = ((Integer) fieldValue).longValue();
                 }
+                else if (fieldValue instanceof BigInteger)
+                {
+                    value = ((BigInteger) fieldValue).longValue();
```

Comment:
can fieldValue == null?, default to 0?

Reply:
that's trapped by the if fieldvalue==null case above it, it'll never be null at this point